readthedocs/projects/static-src/projects/js/import.js

```diff
@@ -0,0 +1,252 @@
+var ko = require('knockout'),
+    $ = require('jquery'),
+    tasks = require('../../../../core/static-src/core/js/tasks'),
+    csrf = require('../../../../core/static-src/core/js/django-csrf');
+
+
+$(function() {
+  var input = $('#id_repo'),
+      repo = $('#id_repo_type');
+
+  input.blur(function () {
+    var val = input.val(),
+        type;
+
+    switch(true) {
+      case /^hg/.test(val):
+        type = 'hg';
+      break;
+
+      case /^bzr/.test(val):
+      case /launchpad/.test(val):
+        type = 'bzr';
+      break;
+
+      case /trunk/.test(val):
+      case /^svn/.test(val):
+        type = 'svn';
+      break;
+
+      default:
+      case /github/.test(val):
+      case /(^git|\.git$)/.test(val):
+        type = 'git';
+      break;
+    }
+
+    repo.val(type);
+  });
+});
+
+function Organization (instance, view) {
+    var self = this;
+    self.id = ko.observable(instance.id);
+    self.name = ko.observable(instance.name);
+    self.slug = ko.observable(instance.slug);
+    self.active = ko.observable(instance.active);
+    self.avatar_url = ko.observable(
+        append_url_params(instance.avatar_url, {size: 32})
+    );
+    self.display_name = ko.computed(function () {
+        return self.name() || self.slug();
+    });
+    self.filtered = ko.computed(function () {
+        var id = view.filter_org();
+        return id && id != self.id();
+    });
+}
+
+function Project (instance, view) {
+    var self = this;
+    self.id = ko.observable(instance.id);
+    self.name = ko.observable(instance.name);
+    self.full_name = ko.observable(instance.full_name);
+    self.description = ko.observable(instance.description);
+    self.vcs = ko.observable(instance.vcs);
+    self.organization = ko.observable();
+    self.html_url = ko.observable(instance.html_url);
+    self.clone_url = ko.observable(instance.clone_url);
+    self.ssh_url = ko.observable(instance.ssh_url);
+    self.matches = ko.observable(instance.matches);
+    self.match = ko.computed(function () {
+        var matches = self.matches();
+        if (matches && matches.length > 0) {
+            return matches[0];
+        }
+    });
+    self.private = ko.observable(instance.private);
+    self.active = ko.observable(instance.active);
+    self.admin = ko.observable(instance.admin);
+    self.is_locked = ko.computed(function () {
+        return (self.private() && !self.admin());
+    });
+    self.avatar_url = ko.observable(
+        append_url_params(instance.avatar_url, {size: 32})
+    );
+
+    self.import_repo = function () {
+        var data = {
+                name: self.name(),
+                repo: self.clone_url(),
+                repo_type: self.vcs(),
+                description: self.description(),
+                project_url: self.html_url(),
+            },
+            form = $('<form />');
+
+        form
+            .attr('action', view.urls.projects_import)
+            .attr('method', 'POST');
```

Comment:
I'm guessing this code will totally not work with JS disabled? Is there a reason to be building the form in JS, instead of having it in the HTML? I guess because we'd need a form for each project?